torchgen/dest/register_dispatch_key.py

```diff
@@ -302,7 +302,7 @@ def gen_structured(self, g: NativeFunctionsGroup) -> List[str]:
                 "Do not explicitly specify Meta dispatch key on structured "
                 "functions, they will be automatically generated for you"
             )
-        elif self.backend_index.dispatch_key == DispatchKey.CompositeExplicitAutograd:
+        elif self.backend_index.dispatch_key == DispatchKey.CompositeExplicitAutogradWithMutations:
```

Comment:
All of the codegen logic for generating `functional -> out=` kernels should now be registered to the new key.

I mentioned this in the PR description - I guess technically it would be more correct to register the functional kernels to the `CompositeExplicitAutogradWithMutations` key, and the inplace kernels to the `CompositeExplicitAutograd` key, but it seems harmless to register them both to the new key.